log4j-appender/src/main/java/org/apache/kafka/log4jappender/KafkaLog4jAppender.java

```diff
@@ -248,16 +257,22 @@ protected void append(LoggingEvent event) {
         String message = subAppend(event);
         LogLog.debug("[" + new Date(event.getTimeStamp()) + "]" + message);
         Future<RecordMetadata> response = producer.send(
-            new ProducerRecord<byte[], byte[]>(topic, message.getBytes(StandardCharsets.UTF_8)));
+                new ProducerRecord<>(topic, message.getBytes(StandardCharsets.UTF_8)));
         if (syncSend) {
             try {
-                response.get();
+                getResponse(response);
             } catch (InterruptedException | ExecutionException ex) {
-                throw new RuntimeException(ex);
+                LogLog.debug("Exception while getting response", ex);
+                if (!ignoreExceptions)
+                    throw new RuntimeException(ex);
             }
         }
     }
 
+    void getResponse(Future<RecordMetadata> response) throws InterruptedException, ExecutionException {
```

Comment:
What value does this method add?

Reply:
for testing purposes, I've resolved it in another way